### PR TITLE
Taxes 💸: Add runtime migrations to schema v2

### PR DESCRIPTION
**Pull request by peelar**
## Context:
The migration classes don't map 1:1 the actual implementation of tax providers. 

So `TaxProvidersPrivateMetadataManagerV2` -> `AvataxConnectionRepository` + `TaxJarConnectionRepository`.

The reason for that is that `AvataxConnectionRepository` and `TaxJarConnectionRepository` are just abstractions on a more general idea of tax providers' metadata manager. They have different logic, but in the end, they all write to the same metadata key: `tax-providers-v2`.

## Featuring:
- migration code
- `run-report`,`run-migration`, and ❗ `run-generate-dummy-config` ❗ scripts. The last one wasn't present in [the Emails and Messages migration](https://github.com/saleor/apps/pull/535) I based my PR on. I added it to test the runtime migrations without creating the obsolete config in the UI.

### CODE REVIEW

File: apps/taxes/scripts/migrations/dummy-utils.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+// This class is used to generate dummy config for the app to check if the runtime migrations work as expected.
+class DummyConfigGenerator {
+  private logger: Logger;
+  constructor(private domain: string) {
+    this.logger = createLogger({
+      name: "DummyConfigGenerator",
+    });
+  }
+  private getFileApl = () => {
```
Comment (lkostrowski) [inline apps/taxes/scripts/migrations/dummy-utils.ts:72]: its not exactly fileApl?

File: apps/taxes/scripts/migrations/dummy-utils.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  private generateDummyTaxProvidersConfig = (): TaxProvidersV1 => dummyTaxProvidersV1Config;
+
+  private generateDummyTaxChannelsConfig = (): ChannelsV1 => dummyChannelsV1Config;
+
+  generate = async () => {
+    console.log("Generating dummy config");
+    const apls = await this.getFileApl();
+
+    console.log({ apls }, "Apls retrieved");
```
Comment (lkostrowski) [inline apps/taxes/scripts/migrations/dummy-utils.ts:84]: We have single APL. Apl contains multiple AuthData.

File: apps/taxes/scripts/migrations/dummy-utils.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    const target = apls.find((apl) => apl.domain === this.domain);
```
Comment (lkostrowski) [inline apps/taxes/scripts/migrations/dummy-utils.ts:86]: we should use saleorApiUrl

File: apps/taxes/scripts/migrations/tax-channels-migration-v1-to-v2.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import { TaxChannelsPrivateMetadataManagerV1 } from "./tax-channels-metadata-manager-v1";
+import { TaxChannelsPrivateMetadataManagerV2 } from "./tax-channels-metadata-manager-v2";
+import { TaxChannelsTransformV1toV2 } from "./tax-channels-transform-v1-to-v2";
+
+export class TaxChannelsV1toV2MigrationManager {
+  private logger: Logger;
+  constructor(
+    private metadataManager: SettingsManager,
+    private saleorApiUrl: string,
+    private options: { mode: "report" | "migrate" } = { mode: "migrate" }
```
Comment (lkostrowski) [inline apps/taxes/scripts/migrations/tax-channels-migration-v1-to-v2.ts:12]: better to use "safe" default, which is "report" in case of a mistake

**Review by lkostrowski — APPROVED** [review]
Hard to say if it works based on the PR itself. Please make sure this works in staging first.

File: apps/taxes/scripts/migrations/dummy-utils.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    console.log("Setting dummy configs");
+
+    await taxProvidersManager.setConfig(dummyTaxProvidersConfig);
+    await taxChannelsManager.setConfig(dummyTaxChannelsConfig);
+
+    console.log("Dummy config set");
+  };
+}
+
+// const dummyConfigGenerator = new DummyConfigGenerator("");
```
Comment (krzysztofwolski) [inline apps/taxes/scripts/migrations/dummy-utils.ts:120]: comments ;)

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**